ska_helpers/retry/api.py

```diff
@@ -25,19 +22,28 @@ def __retry_internal(f, exceptions=Exception, tries=-1, delay=0, max_delay=None,
                    fixed if a number, random if a range tuple (min, max)
     :param logger: logger.warning(fmt, error, delay) will be called on failed attempts.
                    default: retry.logging_logger. if None, logging is disabled.
+    :param args: tuple, function args
+    :param kwargs: dict, function kwargs
     :returns: the result of the f function.
     """
     _tries, _delay = tries, delay
     while _tries:
         try:
-            return f()
+            return f(*args, **kwargs)
         except exceptions as e:
             _tries -= 1
             if not _tries:
                 raise
 
             if logger is not None:
-                logger.warning('%s, retrying in %s seconds...', e, _delay)
+                call_args = list(args)
+                for key, val in kwargs.items():
+                    call_args.append(f'{key}={val}')
+                call_args_str = ', '.join(str(arg) for arg in call_args)
+                func_name = getattr(f, '__name__', 'func')
+                func_call = f'{func_name}({call_args_str})'
+                logger.warning(f'WARNING: {func_call} exception: {e}, retrying '
```

Comment:
WARNING ends up being repeated. This belongs in the logging formatter.
```
In [3]: logging.warning('WARNING: ')                                                                                                                          
WARNING:root:WARNING: 
```

Reply:
The default `logger` that gets defined doesn't have this behavior, it just prints the message. This is indeed different from `logging.warning()`.